### PR TITLE
refactor(ApplicationLauncher): Refactor so it no longer uses Dropdown

### LINKED ISSUE #1241: App Launcher
We now have a separate component for the app launcher so we need to update this to no longer use the dropdown component. 

### App Launcher
https://pf-next.com/components/AppLauncher/examples/
### Core PR
https://github.com/patternfly/patternfly-next/pull/1239

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Refactored ApplicationLauncher into its own component. It still uses DropdownItem and dropdown contexts so as not to break client code.

Fixes #1241.

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js [hunk 1]
```diff
@@ -19,33 +22,71 @@ export const propTypes = {
...
   onToggle: PropTypes.func,
   /** Indicates where menu will be alligned horizontally */
-  position: PropTypes.oneOf(Object.values(DropdownPosition)),
+  position: PropTypes.oneOf(Object.values(ApplicationLauncherPosition)),
   /** Adds accessible text to the button. Required for plain buttons */
   'aria-label': PropTypes.string
 };
 
+// seed for the aria-labelledby ID
+let currentId = 0;
```
Comment (jschuler) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js]: Is this incremented correctly when you have multiple of these on the same page? I think you'll want to use the generateId util instead

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherMenu.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    } else {
+      custom
+        ? (this.refsCollection[`item-${nextIndex}`].current.focus &&
+            this.refsCollection[`item-${nextIndex}`].current.focus()) ||
+          ReactDOM.findDOMNode(this.refsCollection[`item-${nextIndex}`].current).focus()
+        : this.refsCollection[`item-${nextIndex}`].focus();
+    }
+  };
+
+  onKeyDown = event => {
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherMenu.js]: Can we make a constant in the helpers/constants files for the string "up", "ArrowUp" etc.  This keys will be used in multiple places.

Can we also create a helper function in our utils.ts file for the keyHandler.  It is used in multi components.  Make sure that you comment that the custom argument is optional.

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js (hunk 1 above)
Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js]: It looks like it's incremented correctly. This is how Dropdown does it. (It's not new code.)

File: packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js]: I think we want to decouple the Dropdown component completely from Dropdown here.  So we want to create a new ApplicationLauncher Item component.  If there is any common functionality that can be reused.  Maybe move it into our helper utilities.

**Comment by tlabaj** [comment]
Follow up issues have been open to 1) 

1. address the refactoring of common code. issue #1374 
2. address refactoring the component to not use dropdownItems #1375

File: packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js [hunk 4]
```diff
@@ -19,33 +23,69 @@ export const propTypes = {
...
+          isOpen && styles.modifiers.expanded,
+          className
+        )}
+        ref={ref => {
+            this.parentRef = ref;}}>
+        {Children.map(
+          <Toggle id={`pf-toggle-id-${randomId}`} aria-label={ariaLabel} onToggle={onToggle}><ThIcon /></Toggle>, oneToggle =>
+            cloneElement(oneToggle, {
+              parentRef: this.parentRef,
+              id: randomId,
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js]: should id here match line 63?

Comment (rebeccaalpert) [reply packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js]: I think so. That's how it was originally.

**Pull request merged**